### DISCUSSION TITLE
Default values for MN nodes

### DISCUSSION THREAD

**Discussion by rbdavid** [BradyAJohnston/MolecularNodes#416] [Q&A] [answered]
Hi, 
Are the default setups/parameter values for all MN nodes stored in the `~/MN_home_dir/assets/MN_data_file` blender session file? For example, if I wanted to change the default coloring of all carbon atoms to be cyan, could I change the associated MN_color_element node carbon color in that blend file to then pass that color to MN_color_element nodes that I use in future projects (assuming the future project doesn't already have a MN_color_element node already placed with different parameters)? 

Thanks for your time!

**Comment by BradyAJohnston** [comment, marked as answer]
changing those defaults will change the defaults for the nodes, yes.

There are currently two places where colors are set up. The default values for when the nodes are used, are defined inside of the `.blend` file. On import, colors are also applied to the mesh itself, based on the atomic number of the atoms. This is so, if a node to specify color isn't used, there will still be a default color to fall back on. Those are defined here: 

https://github.com/BradyAJohnston/MolecularNodes/blob/087e82206962daf2f1d51fdd327ef337ecc2dad9/molecularnodes/color.py#L56-L169

**Reply by rbdavid** [reply]
Maybe I'm being daft. I've changed a smattering of parameters in the `MN_data_file` but they do not get passed on to brand new projects. I have a video demo of the issue but its too large to drop into the discussion. I'll try to get that uploaded to a youtube channel this evening. But in the mean time, here's a screenshot of the discrepancy between what's set in the MN_data_file (left) and what is pulled out for a brand new project (right):
![Screenshot (42)](https://github.com/BradyAJohnston/MolecularNodes/assets/8484541/744d4eee-e6c7-4be3-837e-ed1da69fbc1c)

I've made some changes to the `MN_data_file` Geometry Node Viewer, just for my own organizational purposes (thus the green background for the Color group and the resorting of the nodes). Nevermind all of that though. To demonstrate the issue I'm having, I've changed the default color of the alanine residue in the `MN_color_res_name_peptide` node to hot pink, just cause. I save the changes made to the `MN_data_file` (I know there's an asterisk in the image but I swear, the change was saved). This is shown on the left hand side of the image. 

I then start a brand new project, download a pdb structure, and represent it with Spheres. The pre-set visualization tree for Spheres is loaded. But the MN_color_res_name_peptide node is not present in the Geometry nodes tree. So I go through the dropdown menus to add it to the tree and get the colors present on the right hand side of the image. None of the colors match with the defaults.

**Reply by BradyAJohnston** [reply]
Ah, you may have to change the 'Default Value' for the node, rather than just the current value. To do so, press tab to enter edit mode for the selected node, then on the panel on the right you should be able to see the inputs. When you select an input there will be a default value option.

**Reply by rbdavid** [reply]
Drats! I hadn't thought to go a level deeper. Thanks Brady!

**Reply by BradyAJohnston** [reply]
that trips me up sometimes too when i am creating the nodes

**Reply by rbdavid** [reply]
I've forked the MN repo and will push some changes to default color values to that tomorrow. Generally, I pulled default colors from those used in VMD since that's my native visualization software. I've also made the default colors match with the colors presented in the node group view. As part of this work, I have also done a bit of arranging the NodeStorage space as a whole so that I could rationalize what nodes were available and how they were organized/created. 

I'm not sure if this type of effort is worth a pull request. I kinda screwed up by not pushing the default color fixes before the reorg of the NodeStorage space. I assume you had it organized in a fashion suitable for development.

**Reply by BradyAJohnston** [reply]
The `NodeStorage` and everything inside of the `.blend` file is unfortunately completely opaque to git, and is just tracked as a binary file.

This unfortunately means that inside of the `.blend` file there is no version control, so it's hard to potentially do any kind PRs that involve changing of the `.blend` file, as branching forks can't be merged very easily (or at all, really).

I'd be happy to accept a PR that introduced some colors that are more standard from other programs though, like VMD. The best approach would be for it all to exist as code though. We could introduce an option for the default color palette, and where this can come from.

The best approach to start with would likely be adding the colors to the `data.py` or `color.py` files. When adding the nodes, I could have a wrapper that changes the default values of the node when added, according to the palette. This means that the underlying `.blend` doesn't have to be changed, the colors are set when the node is added.

**Reply by rbdavid** [reply]
Yeah! Default values for colors/parameters being defined in human-readable code files would be great!
